proto/redpanda/api/dataplane/v1alpha2/pipeline.proto

```diff
@@ -299,10 +354,18 @@ message ListPipelinesRequest {
       maximum: 1000
     }
   ];
+
+  // Value of the next_page_token field returned by the previous response.
+  // If not provided, the system assumes the first page is requested.
   string page_token = 3;
 
   message Filter {
-    string display_name = 1;
+    // Substring match on pipeline name. Case-sensitive.
+    string name_contains = 1 [
+      (buf.validate.field).string.pattern = "^[A-Za-z0-9-_ /]+$",
+      (buf.validate.field).string.min_len = 3,
+      (buf.validate.field).string.max_len = 128
+    ];
```

Comment:
This was changed to use `name_contains` as that is the standard that is used in all of dataplane APIs. 
The List RPC implementation already performs this behaviour, just the name of the param was not consistent with the rest of the APIs. Also the param now matches validation rules of the name property from create request.

Reply:
Good point, wondering if it should actually be min_len = 3 since we may want to use this for quicksearch as people type. I believe we don't do this on other filters.

Reply:
Good point. I've removed min_len.